src/query/query_eval.rs

```diff
@@ -0,0 +1,76 @@
+use crate::index::Index;
+use crate::metadata::Packet;
+use crate::query::query_types::*;
+use crate::query::QueryError;
+
+pub fn eval_query(index: Index, query: QueryNode) -> Result<Vec<Packet>, QueryError> {
+    match query {
+        QueryNode::Latest(inner) => eval_latest(index, inner),
+        QueryNode::Lookup(field, value) => eval_lookup(index, field, value),
+    }
+}
+
+fn eval_latest(index: Index, inner: Option<Box<QueryNode>>) -> Result<Vec<Packet>, QueryError> {
```

Comment:
So this always returns a vector of length 1 but has to be a vector to match the return type of `eval_lookup`?